eng/pipelines/jobs/steps/upload-job-artifacts.yml

```diff
@@ -35,6 +35,13 @@ steps:
   continueOnError: true
   condition: always()
 
+- task: DeleteFiles@1
```

Comment:
can you please explain this change?

Reply:
This file is generated as part of creating the RPM packages for RHEL. It's a symlink to a file by a path that only exists in the Docker container. When we try to copy files by glob matching below, it tries to stat this path and fails. We delete this file in MSBuild, but it doesn't get deleted there unless the RPM package build succeeds. Without this change, binlogs fail to get uploaded when RPM packages aren't built.

Reply:
thanks.

Reply:
This is a suggestion I made to try to make the build debuggable even if we run into this situation:

https://github.com/dotnet/core-setup/blob/bff42d61cf44185dfb90dc4b344e11aa136d9f46/src/pkg/packaging/rpm/package.targets#L65-L72

Now that https://github.com/dotnet/core-setup/pull/8767 is in, the direct cause is gone, and I think it'd be nice to remove this workaround step from `upload-job-artifacts.yml` so it doesn't cause confusion when it shows up in all the jobs using this template.

Reply:
Whoops, didn't refresh before commenting. Fine either way.

Reply:
I just added a new PR to remove it: https://github.com/dotnet/core-setup/pull/8791